demos/whisper-base/whisper.js

```diff
@@ -319,16 +348,27 @@ export class Whisper {
                 position_ids[0],
                 this.mask_4d,
             );
-            // modify the kv cache in place
-            cache_update(
-                decoder_input,
-                decoder_cached_output,
-                i,
-                this.max_sequence_length,
-                this.num_init_tokens,
-                position_ids[0],
-                this.dataType,
-            );
+
+          // modify the kv cache in place
+          if (this.iobinding) {
+              for (let i = 0; i < 6; i++) {
+                  decoder_input[`past_key_values.${i}.decoder.key`] =
+                      decoder_cached_output[`updated_present_key_values.${i}.decoder.key`];
+                  decoder_input[`past_key_values.${i}.decoder.value`] =
+                      decoder_cached_output[`updated_present_key_values.${i}.decoder.value`];
```

Comment:
You should release previous ml tensors manually, I suppose the performance would get better if legacy ml tensors are released in time.

FYI. my code snippet to release ml tensor.
![image](https://github.com/user-attachments/assets/ccd76ea7-7ea2-4f46-b397-601a18a27011)

Reply:
Updated the code to release previous ml tensors manually. Maybe it’s because the GPU memory used by whisper demo is not very large, so I didn’t see the performance improvement.